### PR TITLE
switch core-tests to psc-package. fixes #2830

### LINKED ISSUE #2830: test-everything.sh fails
The script in `core-tests` appears to still be using 0.10 libs.

### PR DESCRIPTION
**Pull request by matthewleon**

### CODE REVIEW

File: core-tests/psc-package.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "core-tests",
+  "set": "psc-0.11.1",
+  "source": "https://github.com/purescript/package-sets.git",
+  "depends": [
```
Comment (paf31) [inline core-tests/psc-package.json]: Could you please add the newer core packages here too. I notice `typelevel-prelude` is missing. Thanks!

File: core-tests/psc-package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "core-tests",
+  "set": "psc-0.11.1",
```
Comment (paf31) [inline core-tests/psc-package.json]: I think we can use 0.11.3.

**Comment by paf31** [comment]
Thanks! Maybe we can add this to the test suite later too. We'd need to install `psc-package` though.

File: core-tests/psc-package.json (hunk 1 above)
Comment (matthewleon) [inline core-tests/psc-package.json]: is there a convenient list of these somewhere?

Comment (paf31) [inline core-tests/psc-package.json]: Just those under the `purescript` GitHub organization.

**Comment by matthewleon** [comment]
ready for review again... well, pending a successful build.

**Comment by paf31** [comment]
I'm surprised Travis is working. I thought we would have to download psc-package.

File: core-tests/test-everything.sh [hunk 3]
```diff
@@ -13,21 +13,20 @@ while getopts 'ci' flag; do
...
-yes 1 | bower i
+psc-package update
 
 if [ "$force_recompile" = "true" ] && [ -d "output" ]; then
   echo "Recompiling..."
   rm -r output
 fi
 
-stack exec purs compile 'tests/**/*.purs' 'bower_components/purescript-*/src/**/*.purs'
+stack exec purs compile 'tests/**/*.purs' '.psc-package/psc-0.11.1/*/*/src/**/*.purs'
```
Comment (paf31) [inline core-tests/test-everything.sh]: This could use `psc-package build`.

File: core-tests/test-everything.sh [hunk 4]
```diff
@@ -13,21 +13,20 @@ while getopts 'ci' flag; do
...
 
-stack exec purs docs 'bower_components/purescript-*/src/**/*.purs' > core-docs.md
+stack exec purs docs '.psc-package/psc-0.11.1/*/*/src/**/*.purs' > core-docs.md
```
Comment (paf31) [inline core-tests/test-everything.sh]: You can use `psc-package sources` here.

Comment (matthewleon) [inline core-tests/test-everything.sh]: I would really like to get that to work, but my bash-fu is weak, and I've tried every permutation of `${ $( $[ $@ xargamafoobar` possible to try to get it to work, and it always fails to find the Prelude.

Comment (matthewleon) [inline core-tests/test-everything.sh]: Ah, you mean just in the `docs` line! Got it.

File: core-tests/test-everything.sh (hunk 3 above)
Comment (matthewleon) [inline core-tests/test-everything.sh]: Is there a way to ensure that `psc-package` uses the version of `purs` that we just compiled?

**Comment by matthewleon** [comment]
Travis does not run this script. See https://github.com/purescript/purescript/issues/2837

File: core-tests/test-everything.sh (hunk 3 above)
Comment (hdgarrood) [inline core-tests/test-everything.sh]: I think `stack exec` should do that.

**Comment by matthewleon** [comment]
Also, it seems that upgrading to the 0.11.3 package set and adding those other packages makes it so `purs compile` claims it can't find the Prelude. Joy.

File: core-tests/test-everything.sh (hunk 3 above)
Comment (matthewleon) [inline core-tests/test-everything.sh]: ah good point, will give it a shot now

**Comment by matthewleon** [comment]
Should be good to go again.

**Comment by matthewleon** [comment]
Okay, it's failing to find tons of modules now. There's probably something obvious I'm missing.

File: core-tests/test-everything.sh [hunk 5]
```diff
@@ -13,21 +13,20 @@ while getopts 'ci' flag; do
...
-yes 1 | bower i
+psc-package update
 
 if [ "$force_recompile" = "true" ] && [ -d "output" ]; then
   echo "Recompiling..."
   rm -r output
 fi
 
-stack exec purs compile 'tests/**/*.purs' 'bower_components/purescript-*/src/**/*.purs'
+stack exec psc-package build
```
Comment (paf31) [inline core-tests/test-everything.sh]: I'm confused - why does this one use `stack exec` but the others don't?

Comment (matthewleon) [inline core-tests/test-everything.sh]: At the risk of seeming really dumb, I have to admit I'm not sure what the "one" in your sentence is referring to. That line uses `stack exec` in order to ensure that it runs the recently-compiled version of `purs`, no?

File: core-tests/test-everything.sh (hunk 4 above)
Comment (matthewleon) [inline core-tests/test-everything.sh]: Sadly, it seems that that doesn't work. Which is a separate issue I'll create.

**Comment by matthewleon** [comment]
Will probably try to solve #2842 and then submit a new PR.

**Pull request closed without merging**